IdentityCore/src/controllers/MSIDRequestControllerFactory.m

```diff
@@ -78,19 +104,23 @@ @implementation MSIDRequestControllerFactory
     }
     
 #if TARGET_OS_IPHONE
-    if ([self canUseBrokerOnDeviceWithParameters:parameters])
+    if ([parameters canUseBroker])
     {
-        BOOL brokerInstalled = [self brokerInstalledWithParameters:parameters];
-        
-        if (brokerInstalled)
+        if (@available(iOS 13.0, *))
         {
-            return [[MSIDBrokerInteractiveController alloc] initWithInteractiveRequestParameters:parameters
-                                                                            tokenRequestProvider:tokenRequestProvider
-                                                                              fallbackController:localController
-                                                                                           error:error];
+            __auto_type controller = [[MSIDBrokerExtensionInteractiveTokenRequestController alloc] initWithInteractiveRequestParameters:parameters
+                                                                                                                   tokenRequestProvider:tokenRequestProvider
+                                                                                                                     fallbackController:localController
```

Comment:
should there also be a fallback controller for silent?

Reply:
We are calling silent flow with silent sso flow as a fallback.